quantum_datasets/qchem.py

```diff
@@ -289,6 +289,190 @@ def calc_expec_qwc(coeffs, parts, obs):
 
         return res_dicts
 
+    def get_initial_states(
+        hamiltonian,
+        symbols,
+        geometry,
+        charge,
+        basis_name,
+        init_state_tol=1e-4,
+        state_qualities=[1.0, 0.8, 0.6, 0.4, 0.2, 0.1, 0.01, 0.001],
+    ):
+        """
+        Get initial states for quantum chemistry simulations.
+        This function performs Hartree-Fock (HF) and Complete Active Space Configuration Interaction (CASCI) calculations
+        to determine the initial states for quantum chemistry simulations. It iterates over different spin states to find
+        the true ground state, and builds an initial state from that of varying quality, either by
+        truncating the number of Slater determinants in the perfect ground state description, or
+        by mixing in a relevant fraction of the first excited state, when truncation is not possible.
+        Args:
+            hamiltonian (qml.Hamiltonian): The Hamiltonian of the system.
+            symbols (list of str): List of atomic symbols.
+            geometry (list of list of float): List of atomic coordinates in Bohr.
+            charge (int): The charge of the molecule.
+            basis_name (str): The basis set name.
+            init_state_tol (float, optional): Tolerance for which Slater determinants to keep in initial state preparation.
+                            All determinants with coefficients below this value are set to zero. Default is 1e-4.
+            state_qualities (list of float, optional): List of state qualities for truncation and re-allocation. Default is [1.0, 0.8, 0.6, 0.4, 0.2, 0.1, 0.01, 0.001].
+        Returns:
+            tuple: A tuple containing:
+            - dets (np.ndarray): Array of determinants.
+            - coeffs (np.ndarray): Array of coefficients representing the determinants.
+        Notes:
+            - The function uses PySCF for quantum chemistry calculations and PennyLane for quantum computing operations.
+        """
+
+        def _casci_to_superposition(keys, norbs):
+            new_dets = []
+            for key in keys:
+                new_det = []
+
+                int_a, int_b = key
+                bin_a = bin(int_a)[2:][::-1]
+                bin_b = bin(int_b)[2:][::-1]
+                bin_a += "0" * (norbs - len(bin_a))
+                bin_b += "0" * (norbs - len(bin_b))
+                for ii in range(len(bin_a)):
+                    new_det.append(int(bin_a[ii]))
+                    new_det.append(int(bin_b[ii]))
+
+                new_dets.append(new_det)
+
+            return np.array(new_dets)
+
+        bohr_to_ang = 0.529177210903
+        from pyscf import gto, scf, ao2mo
+        from overlapper.utils import get_mol_attrs
+        from overlapper.state import do_hf, do_casci, casci_state
```

Comment:
Do we need to add overlapper to the requirements.txt? Or say something about where to find it?

Reply:
Added it!